solana-programs/payment-router/programs/payment-router/src/utils.rs

```diff
@@ -0,0 +1,129 @@
+use anchor_lang::prelude::*;
+use anchor_lang::solana_program::{
+  account_info::next_account_info,
+  program::invoke_signed
+};
+use anchor_spl::token::spl_token;
+
+use crate::constant::{
+  SOL_USDC_TOKEN_ADDRESS,
+  SOL_AUDIO_TOKEN_ADDRESS,
+};
+use crate::error::PaymentRouterErrorCode;
+
+// Returns the token mint address for either SOL AUDIO or SOL USDC.
+fn get_token_mint(is_audio: bool) -> String {
+    if is_audio {
+      SOL_AUDIO_TOKEN_ADDRESS.to_string()
+    } else {
+      SOL_USDC_TOKEN_ADDRESS.to_string()
+    }
+}
+
+/**
+ * 1. Verify PDA ownership of the sender token account.
+ * 2. Verify that the passed in token mint matches the sender token mint.
+ */
+pub fn check_sender(
+  sender: AccountInfo,
+  sender_owner: AccountInfo,
+  is_audio: bool
+) -> Result<()> {
+    // 1. Verify PDA ownership of the sender token account.
+    // Note that anchor checks for the program ownership of the 'sender_owner' account,
+    // i.e. that the owner of the sender account is owned by the program.
+    // This is because we use the account macro with seeds and bump for the 'sender_owner'.
+    let sender_data = sender.data.borrow();
+    let sender_account_owner= <anchor_spl::token::spl_token::state::Account as anchor_spl::token::spl_token::state::GenericTokenAccount>
```

Comment:
If possible we should simplify these imports

Reply:
done in upcoming PR